interactive/shell.py

```diff
@@ -145,3 +146,50 @@ def show(self, test_set):
             self.tt._tprint(items)
         else:
             self.err()
+
+    @line_magic
+    def cache(self, line, ident='pytest/interactive'):
+        """Store a set of tests in the pytest cache for retrieval in another
+        session.
+
+        Usage:
+
+            cache: show a summary of names previously stored in the cache.
+
+            cache del <name>: deletes the named entry from the cache.
+
+            cache add <name> <target>: stores the named tests as target name.
+        """
+        cachedict = self.tt.get_cache_dict()
+        if line:
+            tokens = line.split()
+            first = tokens[0]
+
+            if first == 'del':
+                # delete an entry
+                name = tokens[1]
+                self.tt.set_cache_items(name, None)  # delete
+                self.shell.user_ns.pop(name)
+                self.tr.write(
+                    "Deleted cache entry for '{}'\n".format(name))
+                return
+
+            elif first == 'add':
+                # create a new entry
+                name, target = tokens[1:]
+                testset = self.ns_eval(name)
+                self.tt.set_cache_items(target, testset)
+
+                # update the local shell's ns
+                if testset:
+                    self.shell.user_ns[target] = testset
+                self.tr.write(
+                    "Created cache entry for '{}'\n".format(name))
+                return
+
+            self.tt.err("{} is invalid. See %cache? for usage.".format(line))
```

Comment:
Heh, for all the hard times I give you for putting unnecessary quotes around your format strings, here's actually a good place to put some in.